internal/agent/install.go

```diff
@@ -217,66 +217,90 @@ func RunInstall(c *config.Config) error {
 		c.Install.Device = detectDevice()
 	}
 
-	// Load the installation spec from the Config
-	installSpec, err := config.ReadInstallSpecFromConfig(c)
-	if err != nil {
-		return err
-	}
+	// UKI path. Check if we are on UKI AND if we are running off a cd, otherwise it makes no sense to run the install
+	// From the installed system
+	if internalutils.IsUki() && internalutils.UkiBootMode() == internalutils.UkiRemovableMedia {
```

Comment:
can we introduce two different functions for the control flow instead of putting it all in this function?

Reply:
can you rephrase that? I dont get youo you are suggesting sorry. 

Also, this should be removed and only the uki boot mode one used as that internally cheks if its on uki first.

Reply:
sorry my bad, I forget ppl are not in my head LOL

I felt that it would be better to have the uki and non-uki process put in different functions, because the outer function was already big enough and now it increased

if it will be removed soon then no need to change

Reply:
ahh I see!

Will do

Reply:
nicer?